app/src/main/java/com/chess/clock/engine/TimeControlParser.java

```diff
@@ -165,22 +175,32 @@ public static int getLastTimeControlCheckIndex(Context context) {
             return null;
         }
 
-        ArrayList<TimeControl> timeControls = new ArrayList<TimeControl>();
+        ArrayList<TimeControlWrapper> timeControls = new ArrayList<TimeControlWrapper>();
         try {
             JSONObject json = new JSONObject(jsonString);
             JSONArray timeControlsJSONArray = json.getJSONArray("timecontrols");
             for (int i = 0; i < timeControlsJSONArray.length(); i++) {
                 JSONObject timeControlJSON = timeControlsJSONArray.getJSONObject(i);
+                JSONObject timeIncrementPlayerTwoJSONOBject = null;
+                JSONArray timeControlStagesPlayerTwoJSON = null;
 
                 JSONObject timeIncrementJSONOBject = timeControlJSON.getJSONObject("timeincrement");
                 JSONArray timeControlStagesJSON = timeControlJSON.getJSONArray("stages");
 
+                if (timeControlJSON.has("timeincrement2") && timeControlJSON.has("stages2")) {
```

Comment:
Maybe we should pull these strings into constants.  Also, would there be better names than just adding "2"?